### ISSUE TITLE
Function visible extern with globally visible global shouldn't initialize in function

### ISSUE THREAD

**Issue by TheDan64** [immunant/c2rust#113]
```c
unsigned int *get(void) {
    extern unsigned int bar;

    return &bar;
}

unsigned int bar = 1U + 1U;

int main(void) {
    get();
  
    return 1;
}
```

Will become
```rust
#[no_mangle]
pub unsafe extern "C" fn get() -> *mut libc::c_uint {
    // Initialized in run_static_initializers
    #[no_mangle]
    pub static mut bar: libc::c_uint = 0;
    return &mut bar;
}
unsafe extern "C" fn run_static_initializers() {
    bar = 1u32.wrapping_add(1u32)
}
```

but `bar` isn't accessible since the global is only visible to function `get`.

Interestingly, if the function scoped global is static, then it does become globally visible which is maybe what we want for the above:
```c
unsigned int *get(void) {
    static unsigned int bar = 1U + 1U;

    return &bar;
}
```

becomes
```rust
// Initialized in run_static_initializers
static mut bar: libc::c_uint = 0;
#[no_mangle]
pub unsafe extern "C" fn get() -> *mut libc::c_uint { return &mut bar; }
````

Alternatively, I suppose we could add a top level `extern "C" { #[no_mangle] static bar: ...; }` to the first example rust code

**Comment by rinon** [comment]
Huh, yeah that `bar` declaration in the first example shouldn't be inside the function IMO. Fairly sure this is an incorrect translation.

**Comment by TheDan64** [comment]
I'm starting to think this is our usual issue with non canonical decls

**Comment by TheDan64** [comment]
Fixed in 2c31322f0ba15bacbaf29a02531d526b534f885a and 0afcc8d3386891d1d99d9c4f0e7649804d3bd9af
